src/br/analysis/prereq.py

```diff
@@ -0,0 +1,344 @@
+# Free up cache
+import argparse
+import gc
+import os
+import subprocess
+
+import pandas as pd
+import torch
+
+from br.models.compute_features import compute_features
+from br.models.load_models import get_data_and_models
+from br.models.save_embeddings import (
+    get_pc_loss_chamfer,
+    save_embeddings,
+    save_emissions,
+)
+from br.models.utils import get_all_configs_per_dataset
+
+gc.collect()
+torch.cuda.empty_cache()
+
+# Based on the utilization, set the GPU ID
+
+
+def get_gpu_info():
```

Comment:
Wonder if we should use something like this instead: https://github.com/anderskm/gputil

Reply:
Yeah this looks good! This will need to be updated for the MIG partitioned GPUs though!